source/broadband/webpa_notification.c

```diff
@@ -1041,12 +1041,59 @@ static void addNotifyMsgToQueue(NotifyData *notifyData)
 	WalPrint("*****Returned from addNotifyMsgToQueue*****\n");
 }
 
+/*
+ * @brief To delay sync notifications until FR cloud sync is acknowledged by cloud with Test and Set request for max wait period of 60s
+ */
+void FR_CloudSyncCheck()
+{
+	char *cid = NULL;
+	unsigned int cmc = 0;
+	int FR_cloud_sync_completed = false;
+	WDMP_STATUS ret = WDMP_FAILURE;
+
+	for(int i=0;i<12;i++)
+	{
+		ret = getCmcCidValues(&cmc, &cid);
+		if (ret != WDMP_SUCCESS)
+		{
+			WalError("Unable to get dbCID/CMC value in handleNotificationEvents\n");
+		}
+		else if ((strcmp(cid, "0") == 0) && cmc != 512) //To check whether the device is up after factory reset
+		{
+			WalInfo("factory reset cloud sync is in progress, wait 5sec for process sync notifications\n");
```

Comment:
"Factory reset cloud sync is in progress, wait for 5sec to process sync notifications"

Reply:
updated